gatsby-node.js

```diff
@@ -1,17 +1,43 @@
-const path = require(`path`)
+const PLUGIN_NAME = 'gatsby-plugin-meilisearch'
 
-// Creates a `test` page to test that the plugin and the playground are linked together
-exports.createPages = ({ actions }) => {
-  const { createPage } = actions
-  const blogPostTemplate = path.resolve(`src/components/blog-layout.js`)
+const isObject = obj =>
+  obj && typeof obj === 'object' && obj.constructor === Object
 
-  createPage({
-    path: `test`,
-    component: blogPostTemplate,
-    context: {
-      frontmatter: {
-        title: 'Test page',
-      },
-    },
-  })
+const getValidationError = field =>
+  `[${PLUGIN_NAME}] The field ${field} is required in the plugin configuration`
+
+const validatePluginOptions = (queries, host) => {
+  if (!host) {
+    throw getValidationError('"host"')
+  }
+  if (!isObject(queries)) {
+    throw `[${PLUGIN_NAME}] The field "queries" must be of type object and contain the fields "query" and "indexUid"`
+  }
+  if (!queries.query) {
+    throw getValidationError('"query" in the "queries" object')
+  }
+  if (!queries.indexUid) {
+    throw getValidationError('"indexUid" in the "queries" object')
+  }
```

Comment:
```suggestion
  if (!queries.indexUid) {
    throw getValidationError('"indexUid" in the "queries" object')
  }
  if (!queries.query) {
    throw getValidationError('"query" in the "queries" object')
  }
```

just ordering by importance